src/app/features/toasts/use-toast.ts

```diff
@@ -0,0 +1,15 @@
+import { createContext, useContext } from 'react';
+
+interface ToastContextProps {
+  error(message: string): void;
+  info(message: string): void;
+  success(message: string): void;
+  promise(promise: Promise<any>, msgs: { success: string; error: string }): Promise<any>;
+}
+export const ToastContext = createContext<ToastContextProps | null>(null);
+
+export function useToast() {
+  const context = useContext(ToastContext);
+  if (context) return context;
+  throw new Error('useToast must be used within Toasts');
```

Comment:
```suggestion
  if (!context) throw new Error('useToast must be used within Toasts');
  return context;
```

Nit, most common way/elsewhere in codebase we do this way of early return pattern